pkg/kubelet/kubelet_node_status.go

```diff
@@ -554,3 +571,52 @@ func validateNodeIP(nodeIP net.IP) error {
 	}
 	return fmt.Errorf("Node IP: %q not found in the host's network interfaces", nodeIP.String())
 }
+
+// nodeStatusHasChanged compares the original node and current node's status and
+// returns true if any change happens. The heartbeat timestamp is ignored.
+func nodeStatusHasChanged(originalStatus *v1.NodeStatus, status *v1.NodeStatus) bool {
+	if originalStatus == nil || status == nil {
+		return !apiequality.Semantic.DeepEqual(originalStatus, status)
+	}
+
+	if nodeConditionsHaveChanged(originalStatus.Conditions, status.Conditions) {
+		return true
+	}
+
+	originalConditions := originalStatus.Conditions
+	originalStatus.Conditions = nil
+	conditions := status.Conditions
+	status.Conditions = nil
+	defer func() {
+		originalStatus.Conditions = originalConditions
+		status.Conditions = conditions
+	}()
+	return !apiequality.Semantic.DeepEqual(originalStatus, status)
+}
+
+// nodeConditionsHaveChanged compares the original node and current node's
+// conditions and returns true if any change happens. The heartbeat timestamp is
+// ignored.
+func nodeConditionsHaveChanged(originalConditions []v1.NodeCondition, conditions []v1.NodeCondition) bool {
+	if len(originalConditions) != len(conditions) {
+		return true
+	}
+
+	originalConditionsCopy := make([]v1.NodeCondition, len(originalConditions))
+	copy(originalConditionsCopy, originalConditions)
+	conditionsCopy := make([]v1.NodeCondition, len(conditions))
+	copy(conditionsCopy, conditions)
+
+	sort.Slice(originalConditionsCopy, func(i, j int) bool { return originalConditionsCopy[i].Type < originalConditionsCopy[j].Type })
+	sort.Slice(conditionsCopy, func(i, j int) bool { return conditionsCopy[i].Type < conditionsCopy[j].Type })
+
+	replacedheartbeatTime := metav1.Now()
```

Comment:
nit: we do we need Now? why not simply metav1.Time{}  (zero value)?

Reply:
Done